client/settings/game/gameplay-settings.tsx

```diff
@@ -172,39 +176,58 @@ const GameplayRemasteredForm = React.forwardRef<
       </FormContainer>
     </form>
   )
-})
-
-export interface GameplaySettingsProps {
-  localSettings: LocalSettings
-  scrSettings: ScrSettings
-  formRef: React.Ref<SettingsFormHandle>
-  onChange: (values: GameplaySettingsModel) => void
-  onSubmit: (values: GameplaySettingsModel) => void
 }
 
-export default function GameplaySettings({
-  localSettings,
-  scrSettings,
-  formRef,
-  onChange,
-  onSubmit,
-}: GameplaySettingsProps) {
-  const formModel = useMemo(
-    // TODO(tec27): remove cast once Immutable infers types properly
-    () =>
-      ({
-        ...scrSettings.toJS(),
-        visualizeNetworkStalls: localSettings.visualizeNetworkStalls,
-      } as GameplaySettingsModel),
-    [scrSettings, localSettings],
-  )
+export function GameplaySettings() {
+  const dispatch = useAppDispatch()
+  const localSettings = useAppSelector(s => s.settings.local)
+  const scrSettings = useAppSelector(s => s.settings.scr)
+
+  const onValidatedChange = useStableCallback((model: Readonly<GameplaySettingsModel>) => {
+    dispatch(
+      mergeScrSettings(
+        {
+          apmAlertOn: model.apmAlertOn,
+          apmAlertColorOn: model.apmAlertColorOn,
+          apmAlertSoundOn: model.apmAlertSoundOn,
+          apmAlertValue: model.apmAlertValue,
+          apmDisplayOn: model.apmDisplayOn,
+          colorCyclingOn: model.colorCyclingOn,
+          consoleSkin: model.consoleSkin,
+          gameTimerOn: model.gameTimerOn,
+          minimapPosition: model.minimapPosition,
+          showBonusSkins: model.showBonusSkins,
+          selectedSkin: model.selectedSkin,
+          unitPortraits: model.unitPortraits,
+          showTurnRate: model.showTurnRate,
+        },
+        {
+          onSuccess: () => {},
+          onError: () => {},
+        },
+      ),
+    )
+
+    if (model.visualizeNetworkStalls !== localSettings.visualizeNetworkStalls) {
```

Comment:
Yeah, was just a premature optimization. This is the only settings page that has mixed local/scr settings at the moment, so I thought it's unnecessary to update the local settings every time any of the scr settings change.

Especially since this is a dev-only setting. Which reminds me, maybe I should do this only if `DEV_INDICATOR` is `true`? Since that's the only time we render this setting anyway.